src/mesh/unstructured_mesh.C

```diff
@@ -1412,7 +1412,16 @@ UnstructuredMesh::all_second_order_range (const SimpleRange<element_iterator> &
    */
   for (auto & lo_elem : range)
     {
-      // make sure it is linear order
+      // Skip elements in the range that are already SECOND-order.
+      // Ordinarily one does not have a mixture of FIRST and
+      // SECOND-order elements in an all-2D or all-3D mesh, since
+      // FIRST and SECOND-order elements can't be (conforming)
+      // neighbors, but in a mixed-dimension element mesh, this is
+      // possible.
+      if (lo_elem->default_order() == SECOND)
```

Comment:
Should this be `!= FIRST`, since we may have higher-order?

Reply:
The `!= FIRST` case is still handled by the subsequent  `libmesh_error_msg()`. In other words, we only want to `continue` for `SECOND`-order elements, `THIRD`, `FOURTH`, etc. we'll still throw an error.